sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -81,19 +82,20 @@ def __init__(
     ) -> None:
         self.client_connection = client_connection
         self.id = id
-        self._properties = properties
         self.database_link = database_link
         self.container_link = "{}/colls/{}".format(database_link, self.id)
         self._is_system_key: Optional[bool] = None
         self._scripts: Optional[ScriptsProxy] = None
+        if properties:
+            self.client_connection.container_properties_cache[self.container_link] = _set_properties_cache(properties)
 
     def __repr__(self) -> str:
         return "<ContainerProxy [{}]>".format(self.container_link)[:1024]
 
     async def _get_properties(self) -> Dict[str, Any]:
-        if self._properties is None:
-            self._properties = await self.read()
-        return self._properties
+        if self.container_link not in self.client_connection.container_properties_cache:
+            await self.read()
```

Comment:
once read, I guess we can also update the cache?

Reply:
Yes, the cache is updated in the read itself. Whenever a read is called the cache will be updated.